File: app/src/main/java/com/example/apptiaveneno/Entity/Usuario.java
package com.example.apptiaveneno.Entity;

public class Usuario {

    private int IdUsuario;
    private String NombreCompleto;
    private String Correo;
    private String Clave;

    public Usuario(int idUsuario, String nombreCompleto, String correo, String clave) {
        IdUsuario = idUsuario;
        NombreCompleto = nombreCompleto;
        Correo = correo;
        Clave = clave;
    }

    public int getIdUsuario() {
        return IdUsuario;
    }

    public void setIdUsuario(int idUsuario) {
        IdUsuario = idUsuario;
    }

    public String getNombreCompleto() {
        return NombreCompleto;
    }

    public void setNombreCompleto(String nombreCompleto) {
        NombreCompleto = nombreCompleto;
    }

    public String getCorreo() {
        return Correo;
    }

    public void setCorreo(String correo) {
        Correo = correo;
    }

    public String getClave() {
        return Clave;
    }

    public void setClave(String clave) {
        Clave = clave;
    }
}

File: app/src/main/java/com/example/apptiaveneno/Entity/Usuario.kt
package com.example.apptiaveneno.Entity

import android.os.Parcel
import android.os.Parcelable

data class Usuario(
    val idUsuario: Int,
    val nombreCompleto: String,
    val correo: String,
    val clave: String
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readString() ?: "",
        parcel.readString() ?: "",
        parcel.readString() ?: ""
    ) {

    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(idUsuario)
        parcel.writeString(nombreCompleto)
        parcel.writeString(correo)
        parcel.writeString(clave)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Usuario> {
        override fun createFromParcel(parcel: Parcel): Usuario {
            return Usuario(parcel)
        }

        override fun newArray(size: Int): Array<Usuario?> {
            return arrayOfNulls(size)
        }
    }
}

